src/tests/subcommand_test.go

```diff
@@ -0,0 +1,77 @@
+package tests
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+)
+
+// runAndAssertSubcommand executes a subcommand of the "semantifly" command with the provided arguments
+// and asserts that the output contains the specified assertion statement.
+//
+// Parameters:
+//   - subcommand (string): The subcommand to execute.
+//   - assertStatement (string): The expected substring to be present in the command output.
+//   - args ([]string): Additional arguments to pass to the subcommand.
+func runAndAssertSubcommand(subcommand string, assertStatement string, args []string) error {
```

Comment:
I feel like

```
tempFile, err := os.CreateTemp("", "semantifly_test_*.txt")
...
args := []string{tempFile.Name()}
runAndAssertSubcommand("add", "Added file successfully", args); err != nil {
```

is overly complicated and runAndAssertSubcommand is kind of convoluted/overengineered. I would prefer something like ```runAndCheckStdoutContains("semantifly add semantifly_test.txt", "Added file successfully")``` because it's more readable and obvious what it does.

Reply:
I have added the args section to later add the flags for testing the subcommands. In which case we just add to the args list and call the function instead of writing a new command string.

Reply:
Ahh I think I see now, it is weird that the golang exec.Command requires args to be formatted like that.